merkle/compact/tree.go

```diff
@@ -63,27 +69,24 @@ func isPerfectTree(size int64) bool {
 	return size != 0 && (size&(size-1) == 0)
 }
 
-// GetNodeFunc is a function prototype which can look up particular nodes
+// GetNodesFunc is a function prototype which can look up particular nodes
 // within a non-compact Merkle tree. Used by the compact Tree to populate
 // itself with correct state when starting up with a non-empty tree.
-type GetNodeFunc func(depth int, index int64) ([]byte, error)
+type GetNodesFunc func(ids []NodeID) ([][]byte, error)
 
 // NewTreeWithState creates a new compact Tree for the passed in size.
 //
 // This can fail if the nodes required to recreate the tree state cannot be
 // fetched or the calculated root hash after population does not match the
 // expected value.
-
-// getNodeFn will be called a number of times with the coordinates of internal
-// MerkleTree nodes whose hash values are required to initialize the internal
-// state of the compact Tree.  The expectedRoot is the known-good tree root of
-// the tree at size, and is used to verify the correct initial state of the
-// compact Tree after initialisation.
 //
-// TODO(pavelkalinnikov): Make GetNodeFunc get all nodes at once.
-func NewTreeWithState(hasher hashers.LogHasher, size int64, getNodeFn GetNodeFunc, expectedRoot []byte) (*Tree, error) {
+// getNodesFn will be called a number of times with the coordinates of internal
```

Comment:
Isn't this now called once for all nodes?

Reply:
Indeed. Updated the comment.